tasks/Task.cpp

```diff
@@ -229,7 +243,21 @@ void Task::checkFreeBuffer(std::string const &buffer, AcousticConnection const &
     // Only check the first and actual channel.
     if(buffer.size() > acoustic_connection.freeBuffer.at(0))
         // By now, only a message is logged. Let the data be dropped so it will be shown in the output port.
-        RTT::log(RTT::Error) << "Buffer \"" << buffer << "\" is bigger than free transmission buffer. Split your buffer or reduce the rate of transmission." << std::endl;
+        RTT::log(RTT::Error) << "Usbl_evologics Task.cpp. Buffer \"" << UsblParser::printBuffer(buffer) << "\" has size of \"" << buffer.size() << "\" which is bigger than free transmission buffer (\""<< acoustic_connection.freeBuffer.at(0) <<"\"). Split your buffer or reduce the rate of transmission." << std::endl;
+}
+
+// TODO verify
+// Filter possible <+++ATcommand> in raw_data_input
+void Task::filterRawData( std::string const & raw_data_in)
+{
+    // Find "+++"
+    if (raw_data_in.find("+++") != string::npos)
+    {
+        std::string error_msg = "Usbl_evologics Task.cpp. There is the malicious string \"+++\" in raw_data_input. DO NOT use it as it could be interpreted as a command by device.";
+        std::cout << error_msg << std::endl;
+        RTT::log(RTT::Error) << error_msg << std::endl;
+        exception(MALICIOUS_SEQUENCE_IN_RAW_DATA);
```

Comment:
exception() does not throw (it's only a state change), so you will still send the bad message. You could throw just after having called exception().